dot/parachain/availability-store/availability_store.go

```diff
@@ -109,64 +241,248 @@ func (as *AvailabilityStore) loadChunk(candidate common.Hash, validatorIndex uin
 	return &result, nil
 }
 
-// storeChunk stores a chunk in the availability store
-func (as *AvailabilityStore) storeChunk(candidate common.Hash, chunk ErasureChunk) error {
-	meta, err := as.loadMetaData(candidate)
+func (as *AvailabilityStore) writeChunk(batch *AvailabilityStoreBatch, candidate parachaintypes.CandidateHash,
+	chunk ErasureChunk) error {
+	dataBytes, err := json.Marshal(chunk)
+	if err != nil {
+		return fmt.Errorf("marshalling chunk for candidate %v, index %d: %w", candidate, chunk.Index, err)
+	}
+	err = batch.chunkBatch.Put(append(candidate.Value[:], uint32ToBytes(chunk.Index)...), dataBytes)
+	if err != nil {
+		return fmt.Errorf("writing chunk for candidate %v, index %d: %w", candidate, chunk.Index, err)
+	}
+	return nil
+}
+
+func (as *AvailabilityStore) deleteChunk(batch *AvailabilityStoreBatch, candidate parachaintypes.CandidateHash,
```

Comment:
```suggestion
func (as *AvailabilityStore) deleteErasureChunk(batch *AvailabilityStoreBatch, candidate parachaintypes.CandidateHash,
```

Reply:
This name was used for consistency with names used by Parity in Polkadot codebase.